components/topbar/user/src/index.js

```diff
@@ -24,6 +24,33 @@ const BODY_HAS_SCROLL_DISABLED = 'body-has-scroll-disabled'
 const TITLE_CLASS_NAME = 'sui-TopbarUser-title'
 const FLOW_BUTTON_CLASS_NAME = 'sui-TopbarUser-navButton'
 
+const HEADING_TAGS = {
+  h1: 'h1',
+  h2: 'h2',
+  h3: 'h3',
+  h4: 'h4',
+  h5: 'h5',
+  h6: 'h6'
+}
+
+const renderBrand = ({showBrandIcon, linkFactory: Link, brand}) => {
+  const {headingTag: Heading, icon: BrandIcon, image: BrandImage, name, url} = brand
+
+  const asHeading = Boolean(Heading)
+
+  const brandLink = showBrandIcon ? (
+    <Link href={url} className="sui-TopbarUser-brandIcon" title={name}>
+      <BrandIcon />
+    </Link>
+  ) : (
+    <Link href={url} className="sui-TopbarUser-brand" title={name}>
+      {BrandImage ? <BrandImage /> : name}
+    </Link>
+  )
+
+  return asHeading ? <Heading className="sui-TopbarUser-heading">{brandLink}</Heading> : brandLink
```

Comment:
This can affect the default margins of the brand when wrapped into each tag. Is it considered?
(of course it is retro-compatible, I know, just for understanding)